server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -713,14 +713,20 @@ public Iterator<Setting<?>> settings() {
      * <pre>
      * "index.mapping.ignore_above": 256
      * </pre>
+     * <p>
+     * NOTE: The value for `ignore_above` is the _character count_, but Lucene counts
+     * bytes. Here we set the limit to `32766 / 4 = 8191` since UTF-8 characters may
+     * occupy at most 4 bytes.
      */
-    public static final Setting<Integer> IGNORE_ABOVE_SETTING = Setting.intSetting(
-        "index.mapping.ignore_above",
-        Integer.MAX_VALUE,
-        0,
-        Property.IndexScope,
-        Property.ServerlessPublic
-    );
+
+    public static final Setting<Integer> IGNORE_ABOVE_SETTING = Setting.intSetting("index.mapping.ignore_above", settings -> {
+        if (IndexSettings.MODE.get(settings) == IndexMode.LOGSDB
+            && IndexMetadata.SETTING_INDEX_VERSION_CREATED.get(settings).onOrAfter(IndexVersions.ENABLE_IGNORE_ABOVE_LOGSDB)) {
+            return "8191";
```

Comment:
How can we override this for an index with logsdb mode?

Reply:
I see, this is the default value.. Let's add a comment above, or move it to a static helper function for clarity.

Reply:
The

```
settings -> { ... }
```

lambda just determines the default value if no explicit value is provided for the setting.
I will extract the lambda in a method with a descriptive name.